src/hud/newhud.asm

```diff
@@ -193,41 +193,151 @@ DrawNewHUD:
         LDA.w #$9DD3 ; What we wrote over
 RTS
 
+; This table works with the table directly below it (HUDChargeIconTable)
+; and needs to be in sync with it. For example, the 3rd index is set to 40
+; so the 3rd set of icons in the HUDChargeIconTable needs to be set to the
+; values for the left and right tile icons that make up "40" on the HUD.
+
+HUDChargeDamageTable:
+dw #$0014 ; 20
+dw #$001E ; 30
+dw #$0028 ; 40
+dw #$0032 ; 50
+dw #$003C ; 60
+dw #$0046 ; 70
+dw #$0050 ; 80
+dw #$005A ; 90
+
+dw #$0064 ; 100
+dw #$00C8 ; 200
+dw #$012C ; 300
+dw #$0190 ; 400
+dw #$01F4 ; 500
+dw #$0258 ; 600
+dw #$0384 ; 900
+
+dw #$0096 ; 150
+dw #$00FA ; 250
+dw #$01C2 ; 450
+dw #$02EE ; 750
+
+dw #$0078 ; 120
+dw #$008C ; 140
+dw #$00B4 ; 180
+
+dw #$00D2 ; 210
+.end
+
+HUDChargeIconTable:
+; |L.Side||R.Side||Table Indexes||Damage Value|
+dw #$0C70, #$0C6D;|   00 & 02   ||     20     |
+dw #$0C71, #$0C6F;|   04 & 06   ||     30     |
+dw #$0C72, #$0C6F;|   08 & 0A   ||     40     |
+dw #$0C73, #$0C6E;|   0C & 0E   ||     50     |
+dw #$0C74, #$0C6E;|   10 & 12   ||     60     |
+dw #$0C75, #$0C6A;|   14 & 16   ||     70     |
+dw #$0C74, #$0C6F;|   18 & 1A   ||     80     |
+dw #$0C76, #$0C6F;|   1C & 1E   ||     90     |
+
+dw #$0C77, #$0C6F;|   20 & 22   ||    100     |
+dw #$0C78, #$0C6F;|   24 & 26   ||    200     |
+dw #$0C79, #$0C6F;|   28 & 2A   ||    300     |
+dw #$0C7A, #$0C6F;|   2C & 2E   ||    400     |
+dw #$0C7B, #$0C6F;|   30 & 32   ||    500     |
+dw #$0C7C, #$0C6F;|   34 & 36   ||    600     |
+dw #$0C7D, #$0C6F;|   38 & 3A   ||    900     |
+
+dw #$0C7E, #$0C6E;|   3C & 3E   ||    150     |
+dw #$0C7F, #$0C6E;|   40 & 42   ||    250     |
+dw #$0C68, #$0C6E;|   44 & 46   ||    450     |
+dw #$0C69, #$0C6E;|   48 & 4A   ||    750     |
+
+dw #$0C57, #$0C6D;|   4C & 4E   ||    120     |
+dw #$0C2D, #$0C6F;|   50 & 52   ||    140     |
+dw #$0C2E, #$0C6F;|   54 & 56   ||    180     |
+
+dw #$0C2F, #$0C6F;|   58 & 5A   ||    210     |
+.end
+
 ; Routine that draws the charge damage on the HUD
 NewHUDCharge:
         LDA.w HyperBeamFlag : BNE .hyper
                 LDA.w #HUDHealthDigits : STA.b $00
-                JSL.l ExternalLoadChargeDamage
-                CMP.w #0100 : BCS .draw_3
-                        .draw_2
-                        LDX.w #$00BC
-                        JSR.w HUDDrawTwoDigits
-                        LDA.w #$2C0F : STA.l $7EC6C2 ; Blank tile
-                        BRA +
-                .draw_3
-                LDX.w #$00BA
-                JSR.w HUDDrawThreeDigits
-                BRA +
+                JSL.l ExternalLoadChargeDamage ; This subroutine loads the charge damage value into the accumulator.
```

Comment:
Not a blocker and not a stickler for naming conventions but I might call this just `LoadChargeDamage`, or it's not immediately clear what `External` is referring to here.